// lean-modal/jquery.leanModal.js
module.exports = function leanModal(options) {
    var defaults = {
        top: 100,
        overlay: 0.5,
        closeButton: null,
        onClose: function() {},
        onOpen: function() {},
        onClick: function() {}
    };

    if ($('#lean_overlay').length === 0) {
        var overlay = $("<div id='lean_overlay'></div>");

        $("body").append(overlay);
    }

    options =  $.extend(defaults, options);

    return this.each(function() {

        $(this).click(function(event) {
            options.onOpen(event);
        });

        $(this).click(function(e) {

            var modal_id = $(this).attr("href");

            $(modal_id).click(function (event) {
                options.onClick(event);
            });

            $('#lean_overlay').click(function(event) {
                options.onClose(event);
            });

            $(options.closeButton).click(function(event) {
                options.onClose(event);
            });

            $("#lean_overlay").click(function(event) {
                close_modal(modal_id);
            });

            $(options.closeButton).click(function(event) {
                close_modal(modal_id);
            });

            var modal_height = $(modal_id).outerHeight();
            var modal_width = $(modal_id).outerWidth();

            $('#lean_overlay').css({ 'display' : 'block', opacity : 0 });

            $('#lean_overlay').fadeTo(200, options.overlay);

            $(modal_id).css({

                'display' : 'block',
                'position' : 'fixed',
                'opacity' : 0,
                'z-index': 11000,
                'left' : 50 + '%',
                'margin-left' : -(modal_width/2) + "px",
                'top' : options.top + "px"

            });

            $(modal_id).fadeTo(200,1);

            e.preventDefault();
        });
    });

    function close_modal(modal_id){
        $("#lean_overlay").fadeOut(200);

        $(modal_id).css({ 'display' : 'none' });
    }
};
